packages/react-dom-bindings/src/client/ReactDOMFloatClient.js

```diff
@@ -191,10 +189,20 @@ export type FloatRoot = Document | ShadowRoot;
 // global maps of Resources
 const preloadResources: Map<string, PreloadResource> = new Map();
 
+// getRootNode is missing from IE and old jsdom versions
+function getRootNode(container: Container): FloatRoot {
+  // $FlowFixMe[method-unbinding]
+  return typeof container.getRootNode === 'function'
+    ? /* $FlowFixMe[incompatible-return] Flow types this as returning a `Node`,
+       * but it's either a `Document` or `ShadowRoot`. */
+      container.getRootNode()
+    : container.ownerDocument;
+}
```

Comment:
should we feature detect and swap implementations rather than do this for every runtime call?

If we do feature detect we should also update the `getDocumentFromRoot` function to always just return the passed in root